core/src/main/scala/cell/HandlerPool.scala

```diff
@@ -12,11 +12,18 @@ import lattice.{ DefaultKey, Key, Updater }
 import org.opalj.graphs._
 
 import scala.collection.immutable.Queue
+import scala.util.{ Success, Try }
 
 /* Need to have reference equality for CAS.
+ *
+ * quiescenceCellHandlers use NEXTCallbackRunnable, because (a) pool might reach quiescence
+ * repeatedly and (b) cell might not be completed, when quiescence is reached.
  */
-private class PoolState(val handlers: List[() => Unit] = List(), val submittedTasks: Int = 0) {
-  def isQuiescent(): Boolean =
+private class PoolState(
+  val quiescenceHandlers: List[() => Unit] = List(),
+  val quiescenceCellHandlers: List[NextCallbackRunnable[_, _]] = List(),
```

Comment:
Why do we store `NextCallbackRunnable`s, and not just closures like with `quiescenceHandlers`? One thing I'm a bit skeptical about is that `NextCallbackRunnable[_, _]` is a wildcard type, adding complexity which I'm not sure is warranted.